tests/stress-tests.lisp

```diff
@@ -100,7 +100,7 @@
           ;; per iteration.
           :with qvm := nil
           :for seq :below 16
-          :for name := (format nil "TEST_C__~D" (mod seq 8)) ; make sure we repeat names
+          :for name := (format nil "yTEST_C__~D" (mod seq 8)) ; make sure we repeat names
```

Comment:
remember to return this back to normal

Reply:
I'm seeing a heap-exhausted crash here when I run the test suite locally.  Was there a bug here that got resolved?  I think have the latest changes... HEAD is pointing at cb3dddce7c43950daf55112c1a558a4e28e52781

Reply:
This was originally happening for me too, but @stylewarning made my heap size bigger which fixed the issue. 

 `(sb-ext:dynamic-space-size) --> 8589934592`

Reply:
I thought my original `dynamic-space-size` settings were just too small, but if this is going to break for more people then we should probably address this another way,

Reply:
I *think* this might be a result of the fact that `pure-state` will now always allocate an additional `%trail-amplitudes` array in it's `initialize-instance :after` method in `state-representation.lisp` (around line 115).  Previously, a `pure-state-qvm` didn't have any trial amplitudes (only `noisy-qvm` iirc), so this test is now effectively allocating double the memory. As an aside, I think the way `%trial-amplitudes` are initialized by calling `make-lisp-cflonum-vector` is ignoring the "allocation method" the user requested. So in the case of this test, the `amplitudes` are allocated via shared mem, but `%trial-amplitudes` are allocated on the lisp heap.  It seems this behavior was inherited from the original `noisy-qvm`, so maybe that is the intended behavior, idk.

Reply:
Also, fwiw, this fails when run as `make test` from the commandline for me because the Makefile specifies `--dynamic-space-size $(QVM_WORKSPACE)` where `QVM_WORKSPACE` defaults to 2GB.

Reply:
huh, apparently this test is disabled when running in the gitlab ci. I seem to recall someone saying there are issue with shared mem + docker or something.